master/buildbot/www/hooks/github.py

```diff
@@ -230,20 +230,24 @@ def _get_commit_msg(self, repo, sha):
         :param repo: the repo full name, ``{owner}/{project}``.
             e.g. ``buildbot/buildbot``
         '''
+        if not self._token:
+            return 'No message field'
+
         headers = {
-            'User-Agent': 'Buildbot'
+            'Authorization': 'token ' + self._token,
+            'User-Agent': 'Buildbot',
         }
-        if self._token:
-            headers['Authorization'] = 'token ' + self._token
 
         url = '/repos/{}/commits/{}'.format(repo, sha)
         http = yield httpclientservice.HTTPClientService.getService(
             self.master, self.github_api_endpoint, headers=headers,
             debug=self.debug, verify=self.verify)
         res = yield http.get(url)
-        data = yield res.json()
-        msg = data.get('commit', {'message': 'No message field'})['message']
-        return msg
+        if 200 <= res.code < 300:
```

Comment:
we should probably emit some kind of warning in this case, to let the bot owner know that github is failing some of the requests.

Reply:
Done